### PR TITLE
[TESTING_CUSTOMER]Added tests for customer flow 

### PR DESCRIPTION
**Pull request by Alexander1iubchenko**

### CODE REVIEW

File: src/test/java/com/application/bookingservice/service/CustomerServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+        testDto.setLastName(alice.getLastName());
+        when(customerMapper.toResponseDto(alice)).thenReturn(testDto);
+
+        CustomerResponseDto result = customerService.getById(EXISTING_CUSTOMER_ID);
+
+        assertNotNull(result);
+        verify(customerRepository, times(1)).findById(EXISTING_CUSTOMER_ID);
+        verify(customerMapper, times(1)).toResponseDto(alice);
+        assertEquals(alice.getFirstName(), result.getFirstName());
+        assertEquals(alice.getEmail(), result.getEmail());
```
Comment (fedorovychh) [inline src/test/java/com/application/bookingservice/service/CustomerServiceTest.java:103]: assertEquals before verify

File: src/test/java/com/application/bookingservice/service/CustomerServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                EntityNotFoundException.class,
+                () -> customerService.getById(NON_EXISTENT_CUSTOMER_ID)
+        );
+
+        assertEquals(CUSTOMER_NOT_FOUND_MESSAGE
+                + NON_EXISTENT_CUSTOMER_ID, exception.getMessage());
+
+        verify(customerRepository, times(1)).findById(NON_EXISTENT_CUSTOMER_ID);
+        verify(customerMapper, never()).toResponseDto(any());
+        assertTrue(exception.getMessage().contains(String.valueOf(NON_EXISTENT_CUSTOMER_ID)));
```
Comment (fedorovychh) [inline src/test/java/com/application/bookingservice/service/CustomerServiceTest.java:121]: all assertions before verify

File: src/test/resources/db/customer/delete-customers-from-customers-table.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM customers_roles;
+DELETE FROM customers;
```
Comment (fedorovychh) [inline src/test/resources/db/customer/delete-customers-from-customers-table.sql:2]: add empty line

File: src/test/java/com/application/bookingservice/controller/CustomerControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        CustomerResponseDto expected = responseDto;
+        CustomerResponseDto actual = objectMapper.readValue(mvcResult.getResponse()
+                        .getContentAsString(),
+                CustomerResponseDto.class);
+
+        assertEquals(expected, actual,
+                "Expected should be: " + expected
+                + " but was " + actual
+        );
+
```
Comment (fedorovychh) [inline src/test/java/com/application/bookingservice/controller/CustomerControllerTest.java:99]: redudant empty line

File: src/test/java/com/application/bookingservice/controller/CustomerControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                customer.getAuthorities()
+        );
+        SecurityContextHolder.getContext().setAuthentication(authentication);
+    }
+
+    @AfterEach
+    @Sql(scripts = {"classpath:db/customer/delete-customers-from-customers-table.sql"},
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void cleanUpEach() {
+
```
Comment (fedorovychh) [inline src/test/java/com/application/bookingservice/controller/CustomerControllerTest.java:81]: ` void cleanUpEach() {}`

**Review by fedorovychh — APPROVED** [review]

**Review by BohdanTrue — APPROVED** [review]

**Pull request merged**